### ISSUE TITLE
In the mono image, how to transform depth to point could ?

### ISSUE THREAD

**Issue by bright0072876** [mileyan/pseudo_lidar#15]
In the monocular image, how to get the point could from depth image?

**Comment by mileyan** [comment]
I use the pre-trained DORN model. You can download it from https://github.com/hufu6371/DORN .

**Comment by bright0072876** [comment]
DORN just make the first step depth estimation from RGB image to RGB-depth, but not provide generate point cloud.

**Comment by mileyan** [comment]
You can use my code to convert disparity to point clouds. https://github.com/mileyan/pseudo_lidar#convert-the-disparities-to-point-clouds

**Comment by mileyan** [comment]
Update: Please add `--is_depth` in the command.

**Comment by DeriZSY** [comment]
Hi, do we need to do any processing before using the depth generated by DORN to generate pointcloud?
 
I use the depth generated by DORN with https://github.com/hufu6371/DORN/blob/master/demo_kitti.py . 

Juding from the code, the depth is saved to .png, and the result looks well. 
`depth = depth_prediction(args.filename)
depth = depth*256.0
depth = depth.astype(np.uint16)
img_id = args.filename.split('/')
img_id = img_id[len(img_id)-1]
img_id = img_id[0:len(img_id)-4]
if not os.path.exists(args.outputroot):
    os.makedirs(args.outputroot)
cv2.imwrite(str(args.outputroot + '/' + img_id + '_pred.png'), depth)`


![0000000013_depth_pred](https://user-images.githubusercontent.com/25219214/66487032-3f785200-eade-11e9-91d4-41991411f8db.png)

However,  the point cloud generated with the provided code is obviously wrong. Do I need to do some preprocessing with the depth (for example divided by 256) to use it?

**Comment by DeriZSY** [comment]
I solved the problem described above and successfully generated valid pointcloud with depth generated by DORN. 
Some tips:
1. You should use the caffe provided in the DORN repository instead of any latest versions. 
2. If you choose to generate depth by modify the [kitti demo code](https://github.com/hufu6371/DORN/blob/master/demo_kitti.py) (which I think should be the most convenient way), you need to adjust the data type of the depth as indicated in devkit of KITTI Depth by simply adding: `depth = disp_map.astype(np.float) / 256` at [here](https://github.com/mileyan/pseudo_lidar/blob/master/preprocessing/generate_lidar.py#L72) before project depth to point cloud.

**Comment by mileyan** [comment]
Thanks so much. I have update the code.

**Comment by bright0072876** [comment]
Hi, DeriZsy. Just using the depth image to generate the point clouds or need to predict the disparities first? I have already using the DORN caffe version demo code generate the depth image.

**Comment by DeriZSY** [comment]
> Hi, DeriZsy. Just using the depth image to generate the point clouds or need to predict the disparities first? I have already using the DORN caffe version demo code generate the depth image.

Use the depth directly. Notice the 'is-depth' flag [here](https://github.com/mileyan/pseudo_lidar/blob/master/preprocessing/generate_lidar.py#L44) in the code for lidar generation in this repo.

**Comment by bright0072876** [comment]
First move the depth image to predict_disparity folder?

**Comment by DeriZSY** [comment]
> First move the depth image to predict_disparity folder?

plz read the code yourself... then you got all the answers...  brian is a good thing

**Comment by bright0072876** [comment]
when the mono depth image generate the point clouds, each need the camera calibration file. If using common other image out of KITTI there is no camera calibration file, so cannot generate to the point clouds.

**Comment by mileyan** [comment]
> when the mono depth image generate the point clouds, each need the camera calibration file. If using common other image out of KITTI there is no camera calibration file, so cannot generate to the point clouds.

Yes, you need calibration parameters when you generate the point cloud.
